subsys/net/lib/dns/dns_cache.h

```diff
@@ -0,0 +1,100 @@
+/** @file
+ * @brief DNS cache
+ *
+ * An cache holding dns records for faster dns resolving.
+ */
+
+/*
+ * Copyright (c) 2024 Endress+Hauser AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_DNS_CACHE_H_
+#define ZEPHYR_INCLUDE_NET_DNS_CACHE_H_
+
+#include <stdint.h>
+#include <zephyr/net/dns_resolve.h>
+#include <zephyr/kernel.h>
+
+struct dns_cache_entry {
+	char query[DNS_MAX_NAME_LEN];
+	struct dns_addrinfo data;
+	int64_t creation_uptime;
+	uint32_t ttl;
+	bool in_use;
+};
+
+struct dns_cache {
+	size_t size;
+	struct dns_cache_entry *entries;
+	struct k_mutex *lock;
+};
+
+/**
+ * @brief Statically define and initialize a DNS queue.
+ *
+ * The cache can be accessed outside the module where it is defined using:
+ *
+ * @code extern struct dns_cache <name>; @endcode
+ *
+ * @param name Name of the cache.
+ */
+#define DNS_CACHE_DEFINE(name, cache_size)                                                         \
+	K_MUTEX_DEFINE(name##_mutex);                                                              \
+	static struct dns_cache_entry name##_entries[cache_size] = {0};                            \
```

Comment:
No need to initialize to 0 because this is a static variable.

Reply:
Done